### ISSUE TITLE
JKI State Machine Explorer no longer working

### ISSUE THREAD

**Issue by marketz** [JKISoftware/JKI-State-Machine-Editor#105]
I am using LV2017 0.1.f3 32 bit and using JKI state machine on my desktop programming computer.  The right click menu used to work, but no longer is shown.
I do occasionally take the VI and copy it to a laptop that does not have the JKI state machine installed, and sometimes change the code on the laptop .  Is this breaking the JKI State Machine Explorer functions when going back to my desktop computer?

**Comment by jimkring** [comment]
HIi @marketz. Which JKI State Machine package(s) do you have installed. The JKI State Machine Editor package is deprecated, and you'll need to install the latest JKI State Machine package, which includes the right-click menu options.

**Comment by marketz** [comment]
I have uninstalled the JKI State Machine using the JKI Package Manager, and installed version 2018.0.7.45, which did not fix the problem.
Is the 2018 SM compatible with LV 2017?

**Comment by jimkring** [comment]
Yes, JKI SM 2018 is compatible with VIPM 2013 and greater -- the
right-click options should be available in 2015 and greater.

>

**Comment by marketz** [comment]
Jim,
Thanks for the response.
Here is a screen shot of my right click menu:
![image](https://user-images.githubusercontent.com/64147975/80021733-f8194800-84a8-11ea-860c-30e7fcba8957.png)

If I drop another JKI SM on the block diagram, I can right click and see all of the options for that SM.  

Do you know how I can fix this without replacing the existing SM?

**Comment by jimkring** [comment]
Can you show me a screenshot of the full JKI State Machine that's not
working? There's something about it that's preventing the code from
detecting that it's a JKI SM.

**Comment by marketz** [comment]
Jim,
Up front let me say I am in no way proud of the code around the state machine.  In my defense, I hadn't went through the training (which is why I started it Monday).  
I do have experience from 8 years ago with a string state machine, but that wouldn't work with this project because of the priority of some of the things happening - this is a production machine composing of three actuators (one linear) and a scissors lift, in addition to numerous detectors.
Here it is:
![image](https://user-images.githubusercontent.com/64147975/80025299-66acd480-84ae-11ea-96f3-96df21a0cf85.png)
Thanks

**Comment by jimkring** [comment]
Thanks for posting your code @marketz and for the disclaimers. I think that the JKI SM Editor probably won't be able to work with this code, easily, since there have been a lot of modifications to areas of the code that the JKI SM editor expects to look a different way. If you ping me at jki.net/contact-us we could discuss ways we might be able to help you out.

**Comment by marketz** [comment]
Okay, I sent two messages to 'contact us', first one with cell phone number, which doesn't work since I am still at work.  Second one is correct.  
Are there parts that I can cut and paste to bring me back to functional JKI SM explorer?   Now that I learned some of the functions, it would be nice to have it working.
Thanks.

**Comment by kosist** [comment]
I guess it does not detect JKI SM because there are additional case structures in while loop. Once my colleague has added case structure after state machine's case structure, and SM explorer didn't work any more. And I see now many of such case structures along with state machine, so that could be a problem.

**Comment by marketz** [comment]
Thanks
